src/main/kotlin/org/rust/lang/core/stubs/StubImplementations.kt

```diff
@@ -1038,14 +1028,96 @@ class RsMetaItemStub(
     }
 }
 
+class RsBinaryOpStub(
+    parent: StubElement<*>?, elementType: IStubElementType<*, *>,
+    val op: String
+) : StubBase<RsBinaryOp>(parent, elementType) {
+    object Type : RsStubElementType<RsBinaryOpStub, RsBinaryOp>("BINARY_OP") {
+
+        override fun shouldCreateStub(node: ASTNode): Boolean = createStubIfParentIsStub(node)
+
+        override fun serialize(stub: RsBinaryOpStub, dataStream: StubOutputStream) {
+            dataStream.writeUTFFast(stub.op)
+        }
+
+        override fun deserialize(dataStream: StubInputStream, parentStub: StubElement<*>?): RsBinaryOpStub =
+            RsBinaryOpStub(parentStub, this, dataStream.readUTFFast())
+
+        override fun createStub(psi: RsBinaryOp, parentStub: StubElement<*>?): RsBinaryOpStub =
+            RsBinaryOpStub(parentStub, this, psi.op)
+
+        override fun createPsi(stub: RsBinaryOpStub): RsBinaryOp = RsBinaryOpImpl(stub, this)
+    }
+}
+
+class RsLitExprStub(
+    parent: StubElement<*>?, elementType: IStubElementType<*, *>,
+    val type: RsStubLiteralType?,
+    val integerLiteralValue: String?
+) : RsPlaceholderStub(parent, elementType) {
+    object Type : RsStubElementType<RsLitExprStub, RsLitExpr>("LIT_EXPR") {
+
```

Comment:
:thinking: 
```kotlin
override fun shouldCreateStub(node: ASTNode): Boolean = createStubIfParentIsStub(node)
```

Reply:
Yep, it fixes most cases except function tail expr

Reply:
:thinking: :thinking: 
```kotlin
override fun shouldCreateStub(node: ASTNode): Boolean = createStubIfParentIsStub(node) &&
            node.psi.parent.parent !is RsFunction
```

Reply:
🤦
I was completely sure that `RsLitExprStub.Type` is inheritor of `RsPlaceholderStub.Type` and it's already used `createStubIfParentIsStub`
Sorry, my fault